### PR TITLE
Add support for a match keyword

### PR DESCRIPTION
**Pull request by Nemo157**

### CODE REVIEW

**Comment by Nemo157** [comment]
I'm pretty sure that this will not cover every single case yet, and the error messages aren't the greatest (missing `,`'s normally cause a flow on error instead of highlighting the exact issue, at least the errors are generally around the right place though).

But, it does support more stuff than what is tested :smile:. Even using it inside attributes works fine (I'll add a test for that).

**Comment by Nemo157** [comment]
Rebased onto latest master.

File: maud_macros/src/parse.rs [hunk 1]
```diff
@@ -300,6 +312,100 @@ impl<'cx, 'i> Parser<'cx, 'i> {
...
+                    span: sp,
+                    render: self.render.fork(),
+                }.match_bodies());
+                break;
+            },
+            [ref tt, ..] => {
+                self.shift(1);
+                match_var.push(tt.clone());
+            },
+            [] => parse_error!(self, sp, "expected body for this #match"),
```
Comment (lfairy) [inline maud_macros/src/parse.rs]: This should say `@match`

File: maud_macros/src/parse.rs [hunk 2]
```diff
@@ -300,6 +312,100 @@ impl<'cx, 'i> Parser<'cx, 'i> {
...
+            [ref tt @ fat_arrow!(), ..] => {
+                self.shift(1);
+                body.push(tt.clone());
+                break;
+            },
+            [ref tt, ..] => {
+                self.shift(1);
+                body.push(tt.clone());
+            },
+            _ => parse_error!(self, sp, "invalid #match pattern"),
```
Comment (lfairy) [inline maud_macros/src/parse.rs]: Same here

**Comment by lfairy** [comment]
LGTM, thanks!

**Pull request merged**